controllers/metric_controller.go

```diff
@@ -138,11 +142,13 @@ func (r *MetricReconciler) evaluateMetrics(ctx context.Context, t *redskyv1beta1
 }
 
 func (r *MetricReconciler) collectMetrics(ctx context.Context, t *redskyv1beta1.Trial, probeTime *metav1.Time) (*ctrl.Result, error) {
-	// Index the metric definitions
+	// Fetch the experiment
 	exp := &redskyv1beta1.Experiment{}
 	if err := r.Get(ctx, t.ExperimentNamespacedName(), exp); err != nil {
 		return &ctrl.Result{}, err
 	}
+
+	// Index the metric definitions
 	metrics := make(map[string]*redskyv1beta1.Metric, len(exp.Spec.Metrics))
 	for i := range exp.Spec.Metrics {
 		metrics[exp.Spec.Metrics[i].Name] = &exp.Spec.Metrics[i]
```

Comment:
should we do `metricWithDefaults` here to generate the DeepCopy early?

Reply:
How about if I just make this do the deep copy while indexing and fill in the combined defaults/legacy URL in the loop below?

(the point of not doing defaults up here was that there is no point in computing defaults on completed metrics)